internal/service/common/utils/repository.go

```diff
@@ -26,10 +26,26 @@ var ErrNotFound = errors.New("record not found")
 // Following functions are meant to fulfill basic CRUD operations on the database. More complex queries or bulk operations
 // for Insert or Update should be built in the repository files of the specific service and called one of the Execute helper functions.
 
+// QueryFunc is an abstraction to allow passing either a pool or a transaction query function to any of the utilities.
```

Comment:
this could be much cleaner if you create an interface

```
type DBQuery interface {
   QueryFunc func(ctx context.Context, sql string, args ...any) (pgx.Rows, error)
}

```
and then replace db *pgxpool.Pool by DBQuery

No need to create additional functions and have internal ones (first letter lowercase functions)

Reply:
great idea.

Reply:
fixed